test/python/rest/submit_tests.py

```diff
@@ -0,0 +1,212 @@
+            result = job.cancel()
+            self.assertTrue(result, 'cancel() failed for job ' + job.name)
+
+
+    def test_C_submit_with_parameters(self):
+
+        jobConfig = streamsx.topology.context.JobConfig(submission_parameters=applicationParameters, tracing='info')
+
+        jobConfigOverlay = {}
+        jobConfig._add_overlays(jobConfigOverlay)
+
+        logger.warning('submit bundle ' + os.path.basename(applicationBundle))
+        result = self.service.submit_job(applicationBundle, configuration=jobConfigOverlay)
+        self.assertFalse('status_code' in result, 'submit_job() failed, status code and error description:' + str(result))
+        self.assertTrue('name' in result, 'submit_job() failed, no job name returned, result: ' + str(result))
+
+        logger.warning('let the job run a while')
+        time.sleep(runInterval)
+
+        logger.warning('check job ' + result['name'])
+        jobs = self.instance.get_jobs(result['name'])
+        self.assertTrue(len(jobs)>0, 'submit_job() failed, could not find job')
+        for job in jobs:
+            self.assertTrue(result['name']==job.name, 'submit_job() failed, could not find job')
+            self.assertTrue(job.status=='running', 'submit_job() failed, job is not running, status is ' + job.status + ', health is ' + job.health)
+            self.assertTrue(job.health=='healthy', 'submit_job() failed, job is not healthy, status is ' + job.status + ', health is ' + job.health)
+
+        for job in jobs:
+            logger.warning('store logs for job ' + job.name + ' in subdirectory')
+            filename = job.get_application_logs()
+            self.assertTrue(os.path.isfile(filename), 'get_application_logs() failed, no file stored as ' + filename)
+            #filename = job.store_logs('./logs/'+job.name+'.tgz')
+            #self.assertTrue(filename=='./logs/'+job.name+'.tgz', 'store_logs() failed, stored ' + filename + ' instead of ' + './logs/'+job.name+'.tgz')
+            #for pe in job.get_pes():
+            #    logger.warning('store logs for PE ' + pe.id + ' of job ' + job.name + ' in subdirectory')
+            #    filename = pe.store_console_log('./logs/'+job.name+'.PE_'+pe.id+'.log')
+            #    self.assertTrue(filename=='./logs/'+job.name+'.PE_'+pe.id+'.log', 'store_console_log() failed, stored ' + filename + ' instead of ' + './logs/'+job.name+'.PE_'+pe.id+'.log')
+            #    filename = pe.store_application_trace('./logs/'+job.name+'.PE_'+pe.id+'.trace')
+            #    self.assertTrue(filename=='./logs/'+job.name+'.PE_'+pe.id+'.trace', 'store_appliation_trace() failed, stored ' + filename + ' instead of ' + './logs/'+job.name+'.PE_'+pe.id+'.trace')
+
+        for job in jobs:
+            logger.warning('cancel job ' + job.name)
+            result = job.cancel()
+            self.assertTrue(result, 'cancel() failed for job ' + job.name)
+
+
+    @classmethod
+    def tearDownClass(self):
```

Comment:
Please don't stop the service with a test - we run multiple tests against a service.

Reply:
No problem.